### PR TITLE
Add more logging for historical sync leap

### PR DESCRIPTION
**Pull request by alsrdn**
Add some extra debug logging to the historical sync leap process to help with debugging https://github.com/casper-network/casper-node/issues/3598

### CODE REVIEW

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

File: node/src/types/validator_matrix.rs [hunk 1]
```diff
@@ -220,7 +220,7 @@ impl ValidatorMatrix {
         None
     }
 
-    fn read_inner(&self) -> RwLockReadGuard<BTreeMap<EraId, EraValidatorWeights>> {
+    pub(crate) fn read_inner(&self) -> RwLockReadGuard<BTreeMap<EraId, EraValidatorWeights>> {
```
Comment (Fraser999) [inline node/src/types/validator_matrix.rs]: I strongly think we should avoid doing this - providing a way to return (and hence hold) a lock makes it more likely to cause a deadlock.

Whatever functionality is required, we should ensure the validator matrix can provide it without returning a lock.

**Review by Fraser999 — CHANGES_REQUESTED** [review]
One blocker, but otherwise LGTM.

File: node/src/types/validator_matrix.rs (hunk 1 above)
Comment (alsrdn) [reply node/src/types/validator_matrix.rs]: Fixed in https://github.com/casper-network/casper-node/pull/3615/commits/4bfa012fde7102637b1f53b95605ae265b53774f

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**